lib/trino-filesystem/src/test/java/io/trino/filesystem/TestLocations.java

```diff
@@ -71,27 +72,28 @@ public void testInvalidLocationInAppendPath(String location, String exceptionMes
     private static Stream<Arguments> testGetFileNameFromLocationData()
     {
         return Stream.of(
-                Arguments.of("", ""),
-                Arguments.of("test_file", "test_file"),
-                Arguments.of("test>file", "test>file"),
-                Arguments.of("test_dir/", ""),
-                Arguments.of("/test_file.txt", "test_file.txt"),
-                Arguments.of("test_dir/test_file.txt", "test_file.txt"),
-                Arguments.of("/test_dir/test_file.txt", "test_file.txt"),
-                Arguments.of("test_dir /test_file.txt", "test_file.txt"),
-                Arguments.of("test_dir  /test_file.txt", "test_file.txt"),
-                Arguments.of("test_<dir  /test_file.txt", "test_file.txt"),
-                Arguments.of("test_dir/test_dir2/", ""),
-                Arguments.of("s3://test_dir/test_file.txt", "test_file.txt"),
-                Arguments.of("s3://test_dir/test_dir2/test_file.txt", "test_file.txt"),
-                Arguments.of("s3://dir_with_space /test_file.txt", "test_file.txt"),
-                Arguments.of("file://test_dir/test_file", "test_file"),
-                Arguments.of("file:/test_dir/test_file", "test_file"));
+                Arguments.of("", Optional.empty()),
+                Arguments.of("test_file", Optional.of("test_file")),
+                Arguments.of("test>file", Optional.of("test>file")),
+                Arguments.of(" ", Optional.of(" ")),
```

Comment:
I added this one just out of curiosity and noticed that it is (at the time of this writing) considered a valid location. I find it rather strange.